src/equations/NavierStokes_utils.jl

```diff
@@ -34,18 +34,29 @@ This formulation was the one proved best in Syver's paper i.e. DCF.
 """
 create_right_hand_side_with_closure(setup, psolver, closure, st) = function right_hand_side(
         u, p, t)
-    # not sure if we should keep t as a parameter. t is only necessary for the INS functions when having Dirichlet BCs (time dependent)
-    u_INS = NN_padded_to_INS(u, setup)
-    u_nopad = NN_padded_to_NN_nopad(u, setup)
-    u_for_lux = reshape(u_nopad, size(u_nopad)..., 1)
-    u = INS.apply_bc_u(u_INS, t, setup)
-    F = INS.momentum(u, nothing, t, setup)
-    u_lux = Lux.apply(closure, u_for_lux, p, st)[1]
-    FC = sum_NN_nopad_and_INS(u_lux, F, setup)
```

Comment:
So, is this function the problem:?

Reply:
yes but i could not identify the specific line